rules/msk_topic_config.go

```diff
@@ -0,0 +1,145 @@
+type MskTopicConfigRule struct {
+	tflint.DefaultRule
+}
+
+func (r *MskTopicConfigRule) Name() string {
+	return "msk_topic_config"
+}
+
+func (r *MskTopicConfigRule) Enabled() bool {
+	return true
+}
+
+func (r *MskTopicConfigRule) Link() string {
+	return ReferenceLink(r.Name())
+}
+
+func (r *MskTopicConfigRule) Severity() tflint.Severity {
+	return tflint.ERROR
+}
+
+func (r *MskTopicConfigRule) Check(runner tflint.Runner) error {
+	path, err := runner.GetModulePath()
+	if err != nil {
+		return fmt.Errorf("getting module path: %w", err)
+	}
+	if !path.IsRoot() {
+		logger.Debug("skipping child module")
+		return nil
+	}
+
+	resourceContents, err := runner.GetResourceContent(
+		"kafka_topic",
+		&hclext.BodySchema{
+			Attributes: []hclext.AttributeSchema{
+				{Name: "name"},
+				{Name: "replication_factor"},
+			},
+			Blocks: []hclext.BlockSchema{
+				{
+					Type: "config",
+					Body: &hclext.BodySchema{
+						Attributes: []hclext.AttributeSchema{
+							{Name: "retention.ms"},
+							{Name: "compression.type"},
+							{Name: "cleanup.policy"},
+						},
+					},
+				},
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return fmt.Errorf("getting kafka_topic contents: %w", err)
+	}
+
+	for _, topicResource := range resourceContents.Blocks {
+		if err := r.validateTopicConfig(runner, topicResource); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *MskTopicConfigRule) validateTopicConfig(runner tflint.Runner, topic *hclext.Block) error {
+	if err := r.validateReplicationFactor(runner, topic); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+const (
+	replFactorAttrName = "replication_factor"
+	// See [https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/blob/main/rules/msk_topic_config.md#requirements] for explanation.
+	replicationFactorVal = 3
+)
+
+var replFactorFix = fmt.Sprintf("%s = %d", replFactorAttrName, replicationFactorVal)
+
+func (r *MskTopicConfigRule) validateReplicationFactor(runner tflint.Runner, topic *hclext.Block) error {
+	replFactorAttr, hasReplFactor := topic.Body.Attributes[replFactorAttrName]
+	if !hasReplFactor {
+		nameAttr, hasName := topic.Body.Attributes["name"]
+		if !hasName {
+			/*	when no name attribute, we can not issue a fix, as we insert the replication factor after the name */
+			err := runner.EmitIssue(
+				r,
+				fmt.Sprintf("missing replication_factor: it must be equal to '%d'", replicationFactorVal),
+				topic.DefRange,
+			)
+			if err != nil {
+				return fmt.Errorf("emitting issue without fix: no replication factor: %w", err)
+			}
+			return nil
+		}
+
+		err := runner.EmitIssueWithFix(
+			r,
+			fmt.Sprintf("missing replication_factor: it must be equal to '%d'", replicationFactorVal),
+			topic.DefRange,
+			func(f tflint.Fixer) error {
+				return f.InsertTextAfter(nameAttr.Range, "\n"+replFactorFix)
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("emitting issue with fix: no replication factor: %w", err)
+		}
+		return nil
```

Comment:
I think this function could be a bit simpler if this is extracted to a separate function

```suggestion
		r.reportMissingReplicationFactor(runner, topic)
```

Reply:
Yep, sounds good. Done in https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/12/commits/03e2e21328b278a42e45133b94b10d29f95b61e9